test/new-e2e/tests/sysprobe-functional/apmtags_test.go

```diff
@@ -0,0 +1,326 @@
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/windows"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type apmvmSuite struct {
+	e2e.BaseSuite[environments.WindowsHost]
+
+	testspath string
+}
+
+//go:embed fixtures/system-probe.yaml
+var systemProbeConfig string
+
+func TestUSMAutoTaggingSuite(t *testing.T) {
+	suiteParams := []e2e.SuiteOption{e2e.WithProvisioner(awsHostWindows.ProvisionerNoFakeIntake(
+		awsHostWindows.WithAgentOptions(
+			agentparams.WithSystemProbeConfig(systemProbeConfig),
+		),
+	))}
+	if *devMode {
+		suiteParams = append(suiteParams, e2e.WithDevMode())
+	}
+
+	e2e.Run(t, &apmvmSuite{}, suiteParams...)
+}
+
+type usmTaggingFiles struct {
+	jsonFile      string
+	appConfigFile string
+}
+type usmTaggingTest struct {
+	name        string
+	description string
+
+	clientJSONFile  string
+	clientAppConfig string
+
+	defaultFiles usmTaggingFiles
+	siteFiles    map[string]usmTaggingFiles
+
+	appFiles map[string]usmTaggingFiles
+
+	serverSiteName string
+	serverSitePort string
+	targetPath     string
+
+	expectedClientTags []string
+	expectedServerTags []string
+}
+
+var sites = []windows.IISSiteDefinition{
+	{
+		Name:        "TestSite1",
+		BindingPort: "*:8081:",
+		SiteDir:     path.Join("c:", "site1"),
+		Applications: []windows.IISApplicationDefinition{
+			{
+				Name:         "/site1/app1",
+				PhysicalPath: path.Join("c:", "app1"),
+			},
+			{
+				Name:         "/site1/app2",
+				PhysicalPath: path.Join("c:", "app2"),
+			},
+			{
+				Name:         "/site1/app2/nested",
+				PhysicalPath: path.Join("c:", "app2", "nested"),
+			},
+		},
+	},
+	{
+		Name:        "TestSite2",
+		BindingPort: "*:8082:",
+		SiteDir:     path.Join("c:", "site2"),
+	},
+}
+
+func (v *apmvmSuite) SetupSuite() {
+	t := v.T()
+
+	// Get the absolute path to the test assets directory
+	currDir, err := os.Getwd()
+	require.NoError(t, err)
+
+	reporoot, _ := filepath.Abs(filepath.Join(currDir, "..", "..", "..", ".."))
+	kitchenDir := filepath.Join(reporoot, "test", "kitchen", "site-cookbooks")
+	v.testspath = filepath.Join(kitchenDir, "dd-system-probe-check", "files", "default", "tests")
+
+	// this creates the VM.
+	v.BaseSuite.SetupSuite()
+
+	// get the remote host
+	vm := v.Env().RemoteHost
+
+	err = windows.InstallIIS(vm)
+	require.NoError(t, err)
+	// HEADSUP the paths are windows, but this will execute in linux. So fix the paths
+	t.Log("IIS Installed, continuing")
+
+	t.Log("Creating sites")
+	// figure out where we're being executed from.  These paths should be in
+	// native path separators (i.e. not windows paths if executing in ci/on linux)
+
+	_, srcfile, _, ok := runtime.Caller(0)
+	require.True(t, ok)
+	exPath := filepath.Dir(srcfile)
+
+	for idx := range sites {
+		sites[idx].AssetsDir = path.Join(exPath, "assets")
+	}
+
+	err = windows.CreateIISSite(vm, sites)
+	require.NoError(t, err)
+	t.Log("Sites created, continuing")
+}
+
+// site directories are
```

Comment:
Can these be set under the test directory?
If the test is bailed out or crashes before finishing, garbage will be left in the system.  It is easier and safer to delete just the test directory.

Reply:
in the CI infrastructure, the host is always recreated.  So it will always run on a fresh machine.
The setup is intentionally left in place after a failure so that someone (I) can ssh in and diagnose if the problem is with the test setup, especially when developing the tests.